src/NLog/Config/LoggingConfiguration.cs

```diff
@@ -361,10 +369,12 @@ internal void Dump()
         internal void FlushAllTargets(AsyncContinuation asyncContinuation)
         {
             var uniqueTargets = new List<Target>();
-            foreach (var rule in this.LoggingRules)
+            for (int i = 0; i < this.LoggingRules.Count; i++)
```

Comment:
I was meaning to say When another thread is updating the list and you are trying to make a `for` loop using the `this.LoggingRules.Count` wouldn't it cause the same issue which we have currently?